### DISCUSSION TITLE
How to enable bespoke logging in FreeBSD until logging is built

### DISCUSSION THREAD

**Discussion by cookiemonsteruk** [AdguardTeam/AdGuardHome#4279] [Q&A]
From [https://github.com/AdguardTeam/AdGuardHome/issues/4213] we know we don't have logging in FreeBSD and OpenBSD yet as there is for Linux and MacOS. Thre is a potential to be included in v0.108.0.
In the meantime from the documentation [https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration] it could be setup by modifying the `AdGuardHome.yaml`.
Questions:
1. Do I need to create the file first or will adding the directive `log_file` to the config create it if it doens't exist? 
2. In either case, do I need to create the two files so AdGuard can write stderr and stdout to /var/log/AdGuardHome.err and AdGuardHome.out ?

**Comment by cookiemonsteruk** [comment]
One of those I could try and report, which I did.
For 1. I didn't need to create the file first.
2. It seems all logging goes to my file I have defined '/var/log/AdGuard.log' . That's good, there is logging.
Verbose set to true is that, really verbose as expected.

By the way I am using AdGuardHome on a OPNSense firewall v 22.1, not vanilla FreeBSD.

**Reply by ainar-g** [reply]
Re. 2.  The log is a single stream.  The two different files for stderr and stdout on those systems are purely to make sure that we capture all outputs.
